bin/PerlDDS/Process.pm

```diff
@@ -63,9 +63,9 @@ sub WaitKill ($)
 {
   my $self = shift;
   my $timeout = shift;
-  my $ret_value = $self->SUPER::WaitKill($timeout);
+  my @ret_values = $self->SUPER::WaitKill($timeout);
   $self->return_coverage_process();
-  return $ret_value;
+  return @ret_values;
```

Comment:
This changes the result for the existing calls to this function as forcing a array into a scalar takes the last element. This is the issue I was talking about yesterday. For example:

```perl
sub WaitKill_scalar {
  return 1;
}

sub WaitKill_array {
  return 1, 0;
}

my $s = WaitKill_scalar();
print($s, "\n"); # 1
my $a = WaitKill_array();
print($a, "\n"); # 0
```

What I suggested was to do something like this:

```perl
sub WaitKill {
  my $self = shift;
  my $timeout = shift;
  my $opts = shift // {};
  my $signal_ref = $opts->{signal_ref};

  if ($signal_ref) {
    ${$signal_ref} = 10;
  }
}

my $signal;
WaitKill(undef, 20, {signal_ref => \$signal});
print($signal, "\n");

```

Reply:
Yes, I realized that and used `wantarray` function to decide if the caller needs an array or a scalar.

Reply:
Okay, looks like you did that shortly after or maybe even right before I commented.